### DISCUSSION TITLE
Issue with VipComputeAnalysedLoc in Morphologist 2021

### DISCUSSION THREAD

**Discussion by crmn-jm** [brainvisa/brainvisa.github.io#112] [Q&A] [answered]
Dear BrainVISA experts,

I am a new Brainvisa user and I'm working on the Brainvisa 5.0 on a linux system. I have already run the (default) morphologist 2021 pipeline in some databases but I am trying in a new one and the following error appears in the Histogram analysis step:
!! VIP Error: Bad argument ---- !! VIP Error: Execution aborted in function 'VipComputeAnalysedLoc'.

I tried changing some parameters in T1 bias correction (sampling, field_rigidity and mode) but the error remains.

Can someone help me? Thanks!!

Carmen

**Comment by denisri** [comment]
Hello,
Does this problem occur on a single image, or for several ones in the dataset ?
Could you check that the first steps have worked correctly (normalization, AC/PC points, bias correction) ?
Is it possible to have a look at the image (at least a snapshot) to see if it has any visible problem ?
Denis

**Comment by crmn-jm** [comment]
Hi,

This happens with all the images in the database. I attach a screenshot of a view of the MRI. Attached is also a screenshot of the process. Sometimes it gets stuck either directly in bias correction or already in the histogram analysis step.

Thanks,
Carmen

<img width="147" alt="captura" src="https://user-images.githubusercontent.com/84532380/179535949-55277d12-fbd2-4420-b104-650ebd3ffd10.png">
<img width="605" alt="captura2" src="https://user-images.githubusercontent.com/84532380/179536097-abc52a0a-fee1-413e-8da3-7832eb8bc63b.png">

**Comment by denisri** [comment]
Would it be possible to send me an image, so that I can try it ?

**Comment by crmn-jm** [comment]
Hi Denis,

Thank you very much for the offer. As this is a database of real patients, I have spoken to my supervisor so that we can send it to you. He will be the one to provide it to you.

Best regards,
Carmen

**Comment by denisri** [comment]
Do you have a means to send it to me ? I don't like to write email addresses on the web (to avoid spam bots), but I don't really see anther convenient way, so if needed I'll write it here later.
Denis

**Comment by crmn-jm** [comment]
Don't worry, you don't need to write it down, through your published work we can get your mail. Thank you very much!

**Comment by denisri** [comment, marked as answer]
OK, I have seen the image, and actually yes, there is a problem with the bias correction step, which completely fails to determine the region it focuses on. This is due to a signal/noise calculation which compares noise in the center of the image (brain) and near the corner (background). However when the background is completely black (no noise because of a removal of data), then it fails. This is a weakness in Morphologist that we should fix, but hasn't yet.
There is a simple solution to this problem: add a small gaussian noise to the input data. You can do it using the process in brainvisa: "Tools/misc/Add gaussian noise to data": use the raw T1 MRI as both input and output, and set a small noise average (like 10.0).
The you can restart the pipeline (you can skip the orientation step before bias correction), and for me it works perfectly well after that.
Of course there should be a better solution than this trick which noises the data, but in practice it works well, so before we really fix it, it is OK to process this way.
Denis

**Reply by crmn-jm** [reply]
Thank you so much, Denis!  I will do that
